internal_docs/specs/prompts.md

```diff
@@ -0,0 +1,150 @@
+# Prompts
+
+Authors: @mikeldking @anticorrelator
+
+As a user of Phoenix, I want to be able to create, store, and modify the prompts that I use to interact with LLMs. The primary reason for this is to be able to reuse prompts across different use cases and to be able to load them into different LLMs in the playground and in code as needed.
+
+## Terminology
+
+- **Prompts** refer to the message(s) that are passed into the language model.
+- **Prompt Templates** refer a way of formatting information to get the prompt to hold the information you want (such as context and examples) Prompt templates can include placeholders (variables) for things such as examples (e.x. few-shot), outside context (RAG), or any other external data that is needed.
+- **Prompt Type** refer to the different types of prompts that can be used with the language model. For example, Chat, String. The primary prompt type will be Chat as it is the most common.
+
+## Use-cases
+
+A user may want to store a prompt or prompt template to:
+
+- Experiment with different prompts in the playground and test the output of the LLM against different models.
+- Store off prompts so that it can be loaded into a notebook when performing experiments.
+- Share prompts with other users so that they can have a reasonable starting point for new prompts.
+- Use Prompt as a source of truth for prompts in the case that prompts need to be versioned and tracked over time (e.x. beta-testing a new prompt, rolling back to an older version)
+
+In the abstract a prompt template has:
+
+- A human-readable name (identifier)
+- A description / README (markdown)
+- A prompt type (Chat, String, ...)
+- A format (f-string / mustache)
+- A set of revisions to the prompt
+
+In addition a prompt template might have associated model configurations (e.g. the prompt only works under certain parameters). In this case the prompt might have a model configuration associated with it. This includes:
```

Comment:
Not sure I follow this part. When is this association needed?

Reply:
Which model the prompt template works with - e.g. is the prompt template designed to work with gpt-4o or gemini.

Reply:
Another example is the value of `role`, which in the case of a tool result is `tool` for openai, but `user` for anthropic.

Reply:
> Another example is the value of `role`, which in the case of a tool result is `tool` for openai, but `user` for anthropic.

That is a very interesting edge-case. I'd lean towards the more specific (e.g. openai) but it's a tough wrinkle to tackle. Tool uniformity is scary for sure.